pkg/resources/spark/networkpolicy.go

```diff
@@ -73,7 +73,7 @@ func NewClusterDriverNetworkPolicy(sc *dcv1alpha1.SparkCluster) *networkingv1.Ne
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      InstanceObjectName(sc.Name, "driver"),
 			Namespace: sc.Namespace,
-			Labels:    MetadataLabels(sc),
+			Labels:    MetadataAndAdditionalLabels(MetadataLabels(sc), sc.Spec.Master.Labels),
```

Comment:
in theory it doesn't matter which set of extra labels (worker/master) are added onto the `MetadataLabels`, they both contain the same `dominodatalab.com` labels